src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVM.cpp

```diff
@@ -628,6 +625,315 @@ void CompositionalMultiphaseFVM::updatePhaseMobility( ObjectManagerBase & dataGr
+        GEOSX_WARNING( GEOSX_FMT( "Pressure boundary condition not prescribed on set {}", setName ) );
+      }
+
+      // no need to set the number of components here, it was done while checking pressure
+    } );
+
+    // 3. Check composition BC (global component fraction)
+    fsManager.apply( time,
+                     mesh,
+                     "faceManager",
+                     extrinsicMeshData::flow::globalCompFraction::key(),
+                     [&] ( FieldSpecificationBase const & fs,
+                           string const & setName,
+                           SortedArrayView< localIndex const > const &,
+                           Group &,
+                           string const & )
+    {
+      // 3.1 Check pressure, temperature, and record composition bc application
+      integer const comp = fs.getComponent();
+
+      if( bcPresCompStatusMap.count( setName ) == 0 )
+      {
+        bcConsistent = false;
+        GEOSX_WARNING( GEOSX_FMT( "Pressure boundary condition not prescribed on set {}", setName ) );
+      }
+      if( bcTempStatusMap.count( setName ) == 0 )
+      {
+        bcConsistent = false;
+        GEOSX_WARNING( GEOSX_FMT( "Temperature boundary condition not prescribed on set {}. Note that for face boundary conditions, you must provide a temperature", setName ) );
+      }
+      if( comp < 0 || comp >= m_numComponents )
+      {
+        bcConsistent = false;
+        GEOSX_WARNING( GEOSX_FMT( "Invalid component index [{}] in composition boundary condition {}", comp, fs.getName() ) );
+        return; // can't check next part with invalid component id
+      }
+
+      ComponentMask< MAX_NC > & compMask = bcPresCompStatusMap[setName];
+      if( compMask[comp] )
+      {
+        bcConsistent = false;
+        GEOSX_WARNING( GEOSX_FMT( "Conflicting composition[{}] boundary conditions on set {}", comp, setName ) );
+      }
+      compMask.set( comp );
+    } );
+
+    // 3.2 Check consistency between composition BC applied to sets
+    for( auto const & setEntry : bcPresCompStatusMap )
+    {
+      ComponentMask< MAX_NC > const & compMask = setEntry.second;
+      for( integer ic = 0; ic < m_numComponents; ++ic )
+      {
+        if( !compMask[ic] )
+        {
+          bcConsistent = false;
+          GEOSX_WARNING( GEOSX_FMT( "Boundary condition not applied to composition[{}] on set {}", ic, setEntry.first ) );
+        }
+      }
+    }
+  } );
+
+  return bcConsistent;
+}
+
+namespace internal
+{
+string const faceBcLogMessage = string( "CompositionalMultiphaseFVM {}: at time {}s, " )
+                                + string( "the <{}> boundary condition '{}' is applied to the face set '{}' in '{}'. " )
+                                + string( "\nThe total number of target faces (including ghost faces) is {}. " )
+                                + string( "\nNote that if this number is equal to zero, the boundary condition will not be applied on this face set." );
+}
```

Comment:
I'd generally suggest the following (message text unchanged, only code formatting). I hope uncrustify doesn't mind. In future versions `{fmt}` will require the format string to be `constexpr`, so the `string` one might not work. I also removed the namespace name to avoid external linking for the symbol.
```suggestion
namespace
{
char const faceBcLogMessage[] =
  "CompositionalMultiphaseFVM {}: at time {}s, "
  "the <{}> boundary condition '{}' is applied to the face set '{}' in '{}'.\n" 
  "The total number of target faces (including ghost faces) is {}.\n"
  "Note that if this number is equal to zero, the boundary condition will not be applied on this face set.";
}
```

Reply:
Done, I put all the log messages that were in this format (namespace + string) in the flow solvers to the new format.